hello_world/src/hello-world.cpp

```diff
@@ -1,5 +1,5 @@
-#include "hello-world.hpp"
-#include "helloworld/external/cmakepublic/cmake_public.hpp"
+#include <hello-world.hpp>
+#include <external/cmakepublic/cmake_public.hpp>

```

Comment:
```suggestion
#include "helloworld/hello-world.hpp"
#include <cmakepublic/cmake_public.hpp>
```
(usual C/C++ convention is to use double quotes for header files which are part of the current project and angle brackets for header files from other projects). Getting the first line to work will also require adding a namespace directory `helloworld` under the `include` directory. Not sure why it wasn't there to begin with.

As for the remainder of the suggestion:

IIRC `CMakePublic` installs the header file to `CMAKE_INSTALL_PREFIX/include/cmakepublic/cmake_public.hpp` and when you do `find_package(cmakepublic)` the resulting target's include path will be the `CMAKE_INSTALL_PREFIX/include` part (here `CMAKE_INSTALL_PREFIX` is what was passed to `CMakePublic`, and will be different than the `CMAKE_INSTALL_PREFIX` being passed to HelloWorld; also I can't remember off the top of my head whether the include path has the `cmakepublic` directory on it or not). 

I forget exactly what path we were giving `CMakePublic` as an install prefix, but I recommend: `<stage_dir>/CMAKE_INSTALL_PREFIX/lib/helloworld/cmake/external` (here `<stage_dir>` is where we are staging the install of HelloWorld and `CMAKE_INSTALL_PREFIX` is where the user wants the `CMakeSuperBuild` project installed. This means CMakePublic installs the header to `<stage_dir>/CMAKE_INSTALL_PREFIX/lib/helloworld/cmake/external/include/cmakepublic/cmake_public.hpp` and the library to `<stage_dir>/CMAKE_INSTALL_PREFIX/lib/helloworld/cmake/external/lib/libcmake_public.so`. Simply telling HelloWorld it depends on the `CMakePublic` target should get the paths right though.

Disclaimer: I made the executive decision to install the entire dependency to `<stage_dir>/CMAKE_INSTALL_PREFIX/lib/helloworld/cmake/external/`. Down the road what we probably want to do is incorporate a package manger under the hood of CMaize; once we do that, we'll let the package manager worry about these details. For now as long as we systematically install the dependencies it shouldn't be too hard to redirect them when the time for the package manager comes.